File: 0067_add_binary/add_binary.cpp
#include <iostream>
#include <string>

string add_binary(string a, string b)
{
    string res = "";
    int i = a.size() - 1, j = b.size() - 1, c = 0;
    while (i >= 0 || j >= 0 || c == 1)
    {
        c += (i >= 0) ? a[i--] - '0' : 0;
        c += (j >= 0) ? b[j--] - '0' : 0;
        res = char(c % 2 + '0') + res;
        c /= 2;
    }
    return res;
}

File: 0067_add_binary/add_binary.rb
def add_binary(a, b)
    i = a.length - 1
    j = b.length - 1
    res = []
    c = 0
    while i >= 0 || j >= 0 || c > 0
        c += i < 0 ? 0 : a[i].to_i
        c += j < 0 ? 0 : b[j].to_i
        i -= 1
        j -= 1
        res << (c % 2).to_s
        c /= 2
    end
    res.reverse.join
end